### ISSUE TITLE
Ternary select

### ISSUE THREAD

**Issue by StoneCypher** [ellowrath/ClangMan#18]
This code seems much improved to me so far 😄 

I still don't know what's with this optimalSize stuff, but as long as it's here, I can use it to make an example about clarity

```c
if(fstatfs(fd, &fsInfo) == -1) {
    optimalSize = 4 * 1024 * 1024;
}
else {
    optimalSize = fsInfo.f_bsize;
}
```

Things like this can (and maybe should) be written

```c
optimalSize = (fstatfs(fd, &fsInfo) == -1)
  ? 4 * 1024 * 1024    // default
  : fsInfo.f_bsize;
```

So ... I googled it.  And it looks like this is neckbearding to get the correct block size for the operating system to get the fastest scan speed.  This allowed me to add comments:

```c
optimalSize = (fstatfs(fd, &fsInfo) == -1)  // if the OS expresses no block size preference
  ? 4 * 1024 * 1024                         // default to four meg
  : fsInfo.f_bsize;                         // otherwise use what's requested
```

**Comment by ellowrath** [comment]
Yes.  This was definitely a chunk lifted from stackoverflow. In the short term I'm going to compile my 10k words into the binary. But this bit might live on, as I want to add a command line switch to allow loading arbitrary wordlists.
